File: frontend/src/types/recipe.ts
import {Timestamp} from 'firebase/firestore';

export interface Recipe {
    id: string;
    name: string;
    instructions: string;
    createdAt: Timestamp;
    photos: string[];
    ingredients?: RecipeIngredient[];
    nutritionalValues?: NutritionalValues;
    parentRecipeId: string | null;
    authorId?: string;
    isPublic?: boolean;
}

export interface RecipeIngredient {
    id?: string;
    name: string;
    quantity: number;
    unit: string;
    original?: string;
    categoryId?: string;
    hasCustomUnit?: boolean;
}

export interface NutritionalValues {
    calories?: number;
    protein?: number;
    fat?: number;
    carbs?: number;
}

File: backend/src/main/java/com/noisevisionsoftware/vitema/model/recipe/Recipe.java
package com.noisevisionsoftware.vitema.model.recipe;

import com.google.cloud.Timestamp;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Recipe {
    private String id;
    private String name;
    private String instructions;
    private Timestamp createdAt;
    private List<String> photos;
    private List<RecipeIngredient> ingredients;
    private NutritionalValues nutritionalValues;
    private String parentRecipeId;
    private String authorId;
    private boolean isPublic;
}

